packages/kbn-shared-ux-components/src/page_template/solution_nav/solution_nav.tsx

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import './solution_nav.scss';
+
+import React, { useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import {
+  EuiFlyout,
+  EuiSideNav,
+  EuiSideNavItemType,
+  EuiSideNavProps,
+  useIsWithinBreakpoints,
+} from '@elastic/eui';
+
+import classNames from 'classnames';
+import { KibanaSolutionAvatar, KibanaSolutionAvatarProps } from '../../solution_avatar';
+import { KibanaPageTemplateSolutionNavCollapseButton } from './solution_nav_collapse_button';
+
+export type KibanaPageTemplateSolutionNavProps = EuiSideNavProps<{}> & {
+  /**
+   * Name of the solution, i.e. "Observability"
+   */
+  name: KibanaSolutionAvatarProps['name'];
+  /**
+   * Solution logo, i.e. "logoObservability"
+   */
+  icon?: KibanaSolutionAvatarProps['iconType'];
+  /**
+   * Control the collapsed state
+   */
+  isOpenOnDesktop?: boolean;
+  onCollapse?: () => void;
+};
+
+const SIZE = 248;
+
+const setTabIndex = (items: Array<EuiSideNavItemType<{}>>, isHidden: boolean) => {
+  return items.map((item) => {
+    // @ts-ignore-next-line Can be removed on close of https://github.com/elastic/eui/issues/4925
+    item.tabIndex = isHidden ? -1 : undefined;
+    item.items = item.items && setTabIndex(item.items, isHidden);
+    return item;
+  });
+};
+
+/**
+ * A wrapper around EuiSideNav but also creates the appropriate title with optional solution logo
+ */
+export const KibanaPageTemplateSolutionNav = ({
+  name,
+  icon,
+  items,
+  isOpenOnDesktop = true,
+  onCollapse,
+  ...rest
+}: KibanaPageTemplateSolutionNavProps) => {
+  const isSmallerBreakpoint = useIsWithinBreakpoints(['xs', 's']);
+  const isMediumBreakpoint = useIsWithinBreakpoints(['m']);
+  const isLargerBreakpoint = useIsWithinBreakpoints(['l', 'xl']);
+
+  // This is used for both the EuiSideNav and EuiFlyout toggling
+  const [isSideNavOpenOnMobile, setIsSideNavOpenOnMobile] = useState(false);
+  const toggleOpenOnMobile = () => {
+    setIsSideNavOpenOnMobile(!isSideNavOpenOnMobile);
+  };
+
+  const isHidden = isLargerBreakpoint && !isOpenOnDesktop;
+
+  /**
+   * Create the avatar
+   */
+  const solutionAvatar = icon ? <KibanaSolutionAvatar iconType={icon} name={name} /> : null;
```

Comment:
Needs a custom class name that we have in the styles to add a margin

```suggestion
  const solutionAvatar = icon ? <KibanaSolutionAvatar className="kbnPageTemplateSolutionNavAvatar" iconType={icon} name={name} /> : null;
```